### PR TITLE
.addE step doesn't need and shouldn't have `<:< Vertex` constraint

### PR DESCRIPTION
**Pull request by mpollmeier**
re https://gitter.im/mpollmeier/gremlin-scala?at=6019f4d584e66b7f7eca582b

### CODE REVIEW

File: gremlin-scala/src/test/scala/gremlin/scala/TraversalSpec.scala [hunk 1]
```diff
@@ -871,11 +871,20 @@ class TraversalSpec extends WordSpec with Matchers {
         val michael = g.addV(Person).property(Name -> "michael").head
         val karlotta = g.addV(Person).property(Name -> "karlotta").head
 
-        g.V(michael).addE("loves").to(karlotta).head
-        g.addE("knows").from(michael).to(karlotta).head
+        g.V(michael).addE("loves").to(karlotta).iterate()
```
Comment (jbward1) [inline gremlin-scala/src/test/scala/gremlin/scala/TraversalSpec.scala:874]: Just a QQ on why you're switching from head -> iterate()?

File: gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala [hunk 2]
```diff
@@ -954,16 +954,14 @@ class GremlinScala[End](val traversal: GraphTraversal[_, End]) {
...
-      implicit ev: End <:< Vertex): GremlinScala.Aux[Edge, Labels] = {
+  def addE(label: String, properties: KeyValue[_]*): GremlinScala.Aux[Edge, Labels] = {
     val newTrav = properties.foldLeft(traversal.addE(label)) { (trav, prop) =>
       trav.property(prop.key.name, prop.value)
     }
     GremlinScala[Edge, Labels](newTrav)
   }
 
-  def addE(label: StepLabel[Vertex], properties: KeyValue[_]*)(
-      implicit ev: End <:< Vertex): GremlinScala.Aux[Edge, Labels] =
```
Comment (jbward1) [inline gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala:966]: 👍

File: gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala [hunk 3]
```diff
@@ -954,16 +954,14 @@ class GremlinScala[End](val traversal: GraphTraversal[_, End]) {
...
+  def addE(label: StepLabel[Vertex], properties: KeyValue[_]*): GremlinScala.Aux[Edge, Labels] =
```
Comment (jbward1) [inline gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala:964]: I found the test for `def addE(label: String,..` but not one for `addE(label: StepLabel[Vertex],..`

Is it worth adding a test for the addE StepLabel variant also, or would that be overkill?

File: gremlin-scala/src/test/scala/gremlin/scala/TraversalSpec.scala (hunk 1 above)
Comment (mpollmeier) [reply gremlin-scala/src/test/scala/gremlin/scala/TraversalSpec.scala:874]: it's doing the same thing (executing the traversal), but we don't care about it's results

File: gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala (hunk 3 above)
Comment (mpollmeier) [reply gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala:964]: you're right, i was just being lazy here. good thing that you brought that up, the StepLabel should be `StepLabel[Edge]` here and not `StepLabel[Vertex]`. fixed

**Pull request merged**